packages/shared/src/components/plus/GiftPlusModal.tsx

```diff
@@ -0,0 +1,221 @@
+import { useQuery } from '@tanstack/react-query';
+import type { ModalProps } from '../modals/common/Modal';
+import { Modal } from '../modals/common/Modal';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import CloseButton from '../CloseButton';
+import { ButtonSize, ButtonVariant } from '../buttons/common';
+import { Button } from '../buttons/Button';
+import { PlusTitle } from './PlusTitle';
+import {
+  PaymentContextProvider,
+  usePaymentContext,
+} from '../../contexts/PaymentContext';
+import { plusUrl } from '../../lib/constants';
+import { TextField } from '../fields/TextField';
+import { UserIcon } from '../icons';
+import { gqlClient } from '../../graphql/common';
+import { RECOMMEND_MENTIONS_QUERY } from '../../graphql/comments';
+import { RecommendedMention } from '../RecommendedMention';
+import { BaseTooltip } from '../tooltips/BaseTooltip';
+import type { UserShortProfile } from '../../lib/user';
+import useDebounceFn from '../../hooks/useDebounceFn';
+import { ProfileImageSize, ProfilePicture } from '../ProfilePicture';
+
+interface SelectedUserProps {
+  user: UserShortProfile;
+  onClose: () => void;
+}
+
+const SelectedUser = ({ user, onClose }: SelectedUserProps) => {
+  const { username, name } = user;
+
+  return (
+    <div className="flex w-full max-w-full flex-row items-center gap-2 rounded-10 bg-surface-float p-2">
+      <ProfilePicture user={user} size={ProfileImageSize.Medium} />
+      <span className="flex w-full flex-1 flex-row items-center gap-2 truncate">
+        <Typography bold type={TypographyType.Callout}>
+          {name}
+        </Typography>
+        <Typography
+          type={TypographyType.Footnote}
+          color={TypographyColor.Secondary}
+        >
+          {username}
+        </Typography>
+      </span>
+      <CloseButton type="button" onClick={onClose} size={ButtonSize.XSmall} />
+    </div>
+  );
+};
+
+interface GiftPlusModalProps extends ModalProps {
+  preSelected?: UserShortProfile;
+}
+
+export function GiftPlusModalComponent({
+  preSelected,
+  ...props
+}: GiftPlusModalProps): ReactElement {
+  const [overlay, setOverlay] = useState<HTMLElement>();
+  const { onRequestClose } = props;
+  const { oneTimePayment } = usePaymentContext();
+  const [selected, setSelected] = useState(preSelected);
+  const [index, setIndex] = useState(0);
+  const [query, setQuery] = useState('');
+  const [onSearch] = useDebounceFn(setQuery, 500);
+  const { data: users } = useQuery<UserShortProfile[]>({
+    queryKey: ['search', 'users', query],
+    queryFn: async () => {
+      const result = await gqlClient.request(RECOMMEND_MENTIONS_QUERY, {
+        query,
+      });
+
+      return result.recommendedMentions;
+    },
+    enabled: !!query?.length,
+  });
+  const isVisible = !!users?.length && !!query?.length;
+  const onKeyDown = (e: React.KeyboardEvent) => {
+    const movement = ['ArrowUp', 'ArrowDown', 'Enter'];
```

Comment:
We could use `KeyboardCommand` and `ArrowKey` enums wdyt?

Reply:
Oh, right. Forgot they existed. Let me update it!

Reply:
@sshanzel ping

Reply:
Totally forgot about this 🤦

Reply:
@ilasw should be available now 🚀